include/robot_interfaces/robot_logger.hpp

```diff
@@ -0,0 +1,331 @@
+    {
+
+        int i, j;
+        std::string temp;
+
+        for (i = 0; i < field_name.size(); i++)
+        {
+          if(field_data[i].size() == 1)
+          {
+            temp = field_name[i];
+            header_.push_back(temp);
+          }
+          else
+          {
+            for (j = 0; j < field_data[i].size(); j++)
+            {
+                temp = field_name[i] + " " + std::to_string(j);
+                header_.push_back(temp);
+            }
+          }
+        }
+
+    }
+
+    void append_header_to_file()
+    {
+
+      output_file_.open(output_file_name_, std::ios_base::app);
+      infix_ostream_iterator<std::string> string_iterator(output_file_, ", ");
+
+      header_ = get_header();
+
+      std::copy(header_.begin(), header_.end(), string_iterator);
+      output_file_ << std::endl;
+
+      output_file_.close();
+
+    }
+
+    void append_robot_data_to_file()
+    {
+
+        output_file_.open(output_file_name_, std::ios_base::app);
+
+        std::vector<std::vector<double>> observation_data =
+            observation.get_data();
+        std::vector<std::vector<double>> applied_action_data =
+            applied_action.get_data();
+        std::vector<std::vector<double>> desired_action_data =
+            desired_action.get_data();
+        std::vector<std::vector<double>> status_data = status.get_data();
+
+        for (long int j = index_;
+             j < std::min(index_ + block_size_,
+                          logger_data_->observation->newest_timeindex());
+             j++)
+        {
+            try
+            {
+                applied_action = (*logger_data_->applied_action)[j];
+                desired_action = (*logger_data_->desired_action)[j];
+                observation = (*logger_data_->observation)[j];
+                status = (*logger_data_->status)[j];
+
+                status_data = status.get_data();
+                observation_data = observation.get_data();
+                applied_action_data = applied_action.get_data();
+                desired_action_data = desired_action.get_data();
+
+                output_file_ << logger_data_->observation->timestamp_s(j)
+                             << " , " << j << " , ";
+
+                append_field_data_to_robot_data(status_data);
+                append_field_data_to_robot_data(observation_data);
+                append_field_data_to_robot_data(applied_action_data);
+                append_field_data_to_robot_data(desired_action_data);
+
+                output_file_ << std::endl;
+
+            }
+
+            catch (const std::exception &e)
+            {
+                std::cout << "Trying to access index older than the "
+                             "oldest! Skipping ahead."
+                          << std::endl;
+            }
+        }
+
+        output_file_.close();
+    }
+
+    void append_field_data_to_robot_data(std::vector<std::vector<double>> field_data)
```

Comment:
Shouldn't this be called `append_field_data_to_file`?

Reply:
Yes, you are right.